torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -42,14 +42,16 @@ C10_EXPORT std::vector<at::DeprecatedTypeProperties*> allCUDATypes() {
 namespace {
 const Variable & checked_cast_variable(const Tensor & t, const char * name, int pos) {
   if (!t.defined()) {
-    AT_ERROR("Expected a Tensor of type Variable but found an undefined Tensor for argument #", pos, " '", name, "'");
+    AT_ERROR("Expected a proper Tensor but found an undefined Tensor (or None in python) ",
```

Comment:
nit: proper-> defined? The user is not going to know what a "proper Tensor" is

Reply:
undefined Tensors don't exist on the python side. So "defined" wouldn't mean anything either I think. That's why I mention undefined/None after.

Reply:
Should this be TORCH_CHECK?

Reply:
Also, I'd probably reverse it: "Expected a proper Tensor but got None (or an undefined Tensor in C++)", most of our users are probably going to be going through Python?

Reply:
> Should this be TORCH_CHECK?

Yes I'll do that in a follow up